pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -306,38 +350,89 @@ func determineCandidatesWithMinimalDistanceStrategy(seeds []*gardencorev1beta1.S
 	}
 
 	var (
-		maxMatchingCharacters int
-		shootRegion           = shoot.Spec.Region
-		candidates            []*gardencorev1beta1.Seed
+		// maxMatchingCharacters int
+		minDistance = 1000
+		shootRegion = shoot.Spec.Region
+		candidates  []*gardencorev1beta1.Seed
 	)
 
 	for _, seed := range seeds {
-		if seed.DeletionTimestamp == nil && seed.Spec.Provider.Type == shoot.Spec.Provider.Type && !gardencorev1beta1helper.TaintsHave(seed.Spec.Taints, gardencorev1beta1.SeedTaintInvisible) && common.VerifySeedReadiness(seed) {
+		if seed.DeletionTimestamp == nil && !gardencorev1beta1helper.TaintsHave(seed.Spec.Taints, gardencorev1beta1.SeedTaintInvisible) && common.VerifySeedReadiness(seed) {
 			seedRegion := seed.Spec.Provider.Region
-			var matchingCharacters int
-			for i := 0; i <= len(shootRegion); i++ {
-				prefix := shootRegion[:i]
-				if strings.HasPrefix(seedRegion, prefix) {
+			/*
```

Comment:
Can you remove this old code instead of commenting it out?

Reply:
This was just for discussion of my proposed change :-)